Implement the policy report fields creation flow

rezkiy37 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
The PR introduces a new flow to create a report field. Confirmed translations [here](https://docs.google.com/document/d/1u3JPuDjBrmG9YPthTlQo6OQW4h8uoRhhDegO9z-Oq2c/edit).

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/43684
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

**Note:** the feature works only on the staging server for now.
**Note:** the feature is under the beta (`reportFieldsFeature`).

#### Text type

1. Open the report fields page.
2. Click on "Add field".
3. Select the **"Text"** type.
4. Enter any name.
5. Enter an initial value (_optional_).
6. Click on "Save".
7. Verify that the report field is correctly shown in the list

#### Date type

1. Open the report fields page.
2. Click on "Add field".
3. Enter any name.
4. Select the **"Date"** type.
5. Verify the date row has appeared and the value is "Current date".
6. Click on "Save".
7. Verify that the report field is correctly shown in the list

#### List type

1. Open the report fields page.
2. Click on "Add field".
3. Enter any name.
4. Select the **"List"** type.
5. Verify the list values row has appeared.
6. Click on the row.
7. Verify that the empty content view is shown.
8. Click on "Add value".
9. Create a few more values.
10. Verify that the list values are correctly shown in the list
11. Verify that bulk actions work properly.
12. Click on a value row.
13. Verify that you can edit the value.
14. Go back to the main creation page.
15. Select a default list value.
16. Click on "Save".
17. Verify that the report field is correctly shown in the list.

#### Error cases 

1. Verify that you cannot create a new report field if the same name already exists.
2. Verify that you cannot add a new list value if the same name already exists.

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as "Tests".

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Same as "Tests".

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/5206a0e8-4f46-47aa-86ef-b7131fcd98f0


</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/4142c0f3-70e5-453a-9197-2b6d7e0c4828


</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/bfdd0386-b671-4d9c-939e-c2436e212080


</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/ebdce7be-3920-40f2-8881-6ee0cff4300b


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/877a4b9a-3491-46ec-949d-38ccbeb8837e


</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/57314631/b331068a-0465-4b79-b2a7-935f284d8d3b


https://github.com/Expensify/App/assets/57314631/092ac6f3-34ec-46d3-96a6-41a4372b44dd


</details>

rezkiy37 committed:
Squashed commit of the following:

commit 99eede8ae8698fa6ff646c734012894c1d4e60c8
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 17:02:48 2024 +0200

    fix ts error

commit 931ae8c60f2567fd23795101b533cb92a8069d69
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:37:18 2024 +0200

    fix name

commit 4cecf5323737bf660314cf0ead4b33366c2453f2
Merge: 1a48c84462 2526e31ba9
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:36:35 2024 +0200

    Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow

commit 1a48c84462d7666b6e38ce2727dcaa5633aa87f3
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:30:57 2024 +0200

    integrate list of values

commit 5b33862c7be0af0db6e36046d686c1aff7e52e21
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:08:29 2024 +0200

    integrate add new report field

commit 75ab729e64a284b8022107ba99472e873bec6338
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:07:54 2024 +0200

    create ListValuesSelector

commit 16bb5ac039bbdc9b9929fc41a6cf609b33dbcf0b
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:07:47 2024 +0200

    create ListValuesSelectorModal

commit c80252dd47c85d080ff2f2a2f6da08a8a53891a4
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 16:07:31 2024 +0200

    integrate shouldClearOnClose for TextSelectorModal

commit a5a40614197dd078e12ade29635de904bb5936c7
Merge: 1987234795 5f45b4ccc3
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Mon Jun 24 11:43:14 2024 +0200

    Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow

commit 19872347951fc36f1d3e5ac9d98bce38a13ab8c4
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 18:58:03 2024 +0200

    add rightLabel for date

commit d38ba2ded0e3e51eed03999ee44efcf907ec3ce0
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 16:04:59 2024 +0200

    prepare list values form

commit 473629cf1e5ca2c1e264f77a9743cb4951be3f6b
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:48:38 2024 +0200

    use setInitialCreateReportFieldsForm

commit bfdd2b9501df149afbf335c34771aef3ee0e21da
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:48:20 2024 +0200

    create report fields actions file

commit 19226bc37a0514e81da2738e1d3028d4b85b861a
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:22:57 2024 +0200

    rework date value

commit ed7f7dc53a0d2d4becf5a4413b58f92e9de5aa45
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:22:23 2024 +0200

    add translations

commit 2ffc376af38deb228cfbef21b9c1ddfb0b8a05de
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:12:37 2024 +0200

    clarify comments

commit e1b931f2866831f7f62302802e1817dc750149bc
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:11:38 2024 +0200

    integrate alternative text for type picker

commit 2532fa5fedbb3d7c5b5fc2645bc1a376cf23bb53
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:06:02 2024 +0200

    pass subtitles

commit 2d94a2d1ec0db913aff8a329eb800d6c052da784
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:05:52 2024 +0200

    support subtitle in TypeSelectorPicker

commit 7e3a26c1a6be1a6cb4cb039a0004a34e68a34caa
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 15:05:17 2024 +0200

    add padding for subtitle of TextSelectorModal

commit d52f20e48ab8b07e9b8499bbdd34db0fbbc3d202
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 14:53:49 2024 +0200

    support subtitle in TextPicker

commit a25217f0c06f238e464be23e510c412120058b20
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 14:52:20 2024 +0200

    add translations

commit b8295a42727196cb2945d9a7ed395c7490e1124b
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 14:27:43 2024 +0200

    rename to list

commit 8ea203c908710f7209694e1bc8507dc861b4bfe8
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 14:24:40 2024 +0200

    use list text

commit ca31ef88f0e0bebbd07995831999b6ab9663fe9e
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 12:25:54 2024 +0200

    test

commit 46b7d4acd9ddf87398c5b062e4b3b0902a54ca03
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 12:25:41 2024 +0200

    test

commit 2070feae7934839013a7c089d3f4ee153cf1b854
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Fri Jun 21 12:21:53 2024 +0200

    test

commit 951d7c2bc6fb3bd4eb4da208634145ed254aaa83
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 17:37:12 2024 +0200

    reset initial value on type change

commit bf86638b103538f93703322bff96d084c366ec89
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 17:18:56 2024 +0200

    add input ref

commit cffa2d5d1c2755cab0bcf2acd610451ebd599bb8
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 17:08:19 2024 +0200

    fix refs

commit 9638f48b188400dad7e56313ee8bbb7a2756b252
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 16:59:01 2024 +0200

    fix default value of data picker

commit a21d8b07cfd3139298593dccf19010d911baf797
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 16:46:52 2024 +0200

    integrate DateSelector

commit 8a30dc5aeaa52630ad7f37a269743bd83ca13005
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 16:30:43 2024 +0200

    remove label type

commit 206f3d64bc1771920add496d06ef87a7863daff2
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 15:58:59 2024 +0200

    improve TypeSelector props

commit 6a4342cf3c973252ee369bbd5ec664a20624802b
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 15:52:11 2024 +0200

    add translations

commit 66f55555e0c99ee836bc4b2d2925d516b427e9ab
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 15:12:56 2024 +0200

    rename

commit c6fb5d9a7a1d74c685b873f745069c20707920b2
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 15:09:13 2024 +0200

    integrate draft creation of report field

commit 1ea2d2e0923007690bffe290de77d9ac25d7721b
Merge: d41223ad6a 5bbbb11e15
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 14:56:04 2024 +0200

    Merge branch 'feature/43678-report-fields-on-more-features-page' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow

commit d41223ad6a69f96f5c38f1392ec4b5951e266fdd
Merge: e135c2fef6 526ba21c0d
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Thu Jun 20 14:55:43 2024 +0200

    Merge branch 'feat-43677-report-fields-main-page' of https://github.com/waterim/Expnesify into feature/43684-report-fields-creating-flow

commit e135c2fef638cf84bde2b0cfe66b215635136103
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Wed Jun 19 15:30:09 2024 +0200

    create and connect CreatePolicyReportFieldPage

commit ecd436069654bfdd81d712a77b6b81c5dd70850d
Merge: 1c5af2a407 18999b8172
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Wed Jun 19 15:08:18 2024 +0200

    Merge branch 'feat-43677-report-fields-main-page' of https://github.com/waterim/Expnesify into feature/43684-report-fields-creating-flow

commit 1c5af2a407928b588a12c7c6e4c2a6cda9178e14
Merge: d5d1e04a3a eab52fb963
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Wed Jun 19 14:35:39 2024 +0200

    Merge branch 'feature/43678-report-fields-on-more-features-page' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow

commit d5d1e04a3acf0517a831f30f1013ca9378f82f1f
Merge: 904d49d010 67563a18ca
Author: Mykhailo Kravchenko <rezkiy37@gmail.com>
Date:   Wed Jun 19 14:34:36 2024 +0200

    Merge branch 'feat-43677-report-fields-main-page' of https://github.com/waterim/Expnesify into feature/43684-report-fields-creating-flow

rezkiy37 commented:
Original PR - https://github.com/Expensify/App/pull/44089.

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
integrate WorkspaceListValuesPage

rezkiy37 committed:
rename screen

rezkiy37 committed:
integrate WorkspaceListValuesPage

rezkiy37 committed:
integrate WorkspaceAddValuePage

rezkiy37 committed:
remove horizontal spacing

rezkiy37 committed:
integrate submitting and validation for new value

rezkiy37 committed:
simplify WorkspaceListValuesPage

rezkiy37 committed:
remove list value selector modal

rezkiy37 committed:
integrate value editing

rezkiy37 committed:
integrate value deleting

rezkiy37 committed:
add empty state

rezkiy37 committed:
add right label for values

rezkiy37 committed:
use value index instead of name

rezkiy37 committed:
fix value deleting

rezkiy37 committed:
integrate InitialListValueSelector

rezkiy37 committed:
reset ios changes

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
fix nav settings

rezkiy37 committed:
use transparent bg for empty list

rezkiy37 committed:
remove right carets for lists

rezkiy37 committed:
fix string

rezkiy37 committed:
remove right spacing for header

rezkiy37 committed:
integrate bulk actions for values

rezkiy37 committed:
add translations

rezkiy37 committed:
move translations

rezkiy37 committed:
move component

rezkiy37 committed:
update comments

rezkiy37 committed:
remove comment

rezkiy37 committed:
use default values

rezkiy37 committed:
integrate validateReportFieldListValueName

rezkiy37 committed:
integrate validation for create report field page

rezkiy37 committed:
integrate draft form submission

jamesdeanexpensify committed:
Update en.ts

jamesdeanexpensify committed:
Update en.ts

jamesdeanexpensify committed:
Update en.ts

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
integrate createReportField and update imports

rezkiy37 committed:
centrilize list value empty

rezkiy37 committed:
sync text up

mountiny commented on src/CONST.ts:
```diff
@@ -1742,6 +1742,11 @@ const CONST = {
             DISABLE: 'disable',
             ENABLE: 'enable',
         },
+        REPORT_FIELDS_VALUES_BULK_ACTION_TYPES: {
+            DELETE: 'delete',
+            DISABLE: 'disable',
+            ENABLE: 'enable',
+        },
```
I feel like we would want to merge these together to some generic BULK_ACTION_TYPE eventuallly

rezkiy37 replied:
So, I am going to do it in a scope of this PR.

rezkiy37 replied:
Done - 95856f1d8b804d02a3ec46c936bb8a08a8083da5.

rezkiy37 committed:
created first test scenario

fabioh8010 commented on src/components/TextPicker/TextSelectorModal.tsx:
```diff
@@ -59,12 +69,13 @@ function TextSelectorModal({value, description = '', onValueSelected, isVisible,
                     contentContainerStyle={[styles.flex1, styles.mh5, styles.mb5]}
                     keyboardShouldPersistTaps="handled"
                 >
+                    <View style={styles.pb4}>{!!subtitle && <Text style={StyleUtils.combineStyles([styles.sidebarLinkText, styles.optionAlternateText, styles.pre])}>{subtitle}</Text>}</View>
```
Do we need really to use `StyleUtils.combineStyles()` here? Have you tested without it?

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+    priorValueName: string,
+    listValues: string[],
+    inputID: InputID,
+): FormInputErrors<typeof ONYXKEYS.FORMS.WORKSPACE_REPORT_FIELDS_FORM> {
+    const errors: FormInputErrors<typeof ONYXKEYS.FORMS.WORKSPACE_REPORT_FIELDS_FORM> = {};
```
```suggestion
): FormInputErrors<WorkspaceReportFieldsForm> {
    const errors: FormInputErrors<WorkspaceReportFieldsForm> = {};
```

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+        errors[inputID] = Localize.translateLocal('workspace.reportFields.listValueRequiredError');
+    } else if (priorValueName !== valueName && listValues.some((currentValueName) => currentValueName === valueName)) {
+        errors[inputID] = Localize.translateLocal('workspace.reportFields.existingListValueError');
+    } else if ([...valueName].length > CONST.WORKSPACE_REPORT_FIELD_POLICY_MAX_LENGTH) {
```
```suggestion
    } else if (valueName.length > CONST.WORKSPACE_REPORT_FIELD_POLICY_MAX_LENGTH) {
```

rezkiy37 replied:
It is a common pattern in the project. They explain this: 
> Uses the spread syntax to count the number of Unicode code points instead of the number of UTF-16 code units.

<img width="377" alt="Screenshot 2024-06-28 at 12 25 54" src="https://github.com/Expensify/App/assets/57314631/c4b17bcd-8af0-47e9-aa01-3bf99e33af3d">

fabioh8010 replied:
Oh.. didn't know that, thanks!

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+import * as ValidationUtils from './ValidationUtils';
+
+/**
+ * Gets the translation key for the report field type
```
```suggestion
 * Gets the translation key for the report field type.
```

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+}
+
+/**
+ * Gets the translation key for the alternative text for the report field
```
```suggestion
 * Gets the translation key for the alternative text for the report field.
```

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+}
+
+/**
+ * Validates the list value name
```
```suggestion
 * Validates the list value name.
```

fabioh8010 commented on src/libs/WorkspaceReportFieldsUtils.ts:
```diff
@@ -0,0 +1,68 @@
+    return errors;
+}
+
+function generateFieldID(name: string) {
```
Could you add description for this one?

fabioh8010 commented on src/libs/actions/Policy/ReportFields.ts:
```diff
@@ -0,0 +1,185 @@
+        reportFields: JSON.stringify([newReportField]),
+    };
+
+    // eslint-disable-next-line no-console
+    console.group('createReportField');
+    // eslint-disable-next-line no-console
+    console.log('parameters', parameters);
+    // eslint-disable-next-line no-console
+    console.log('onyxData', onyxData);
+    // eslint-disable-next-line no-console
+    console.groupEnd();
```
👀

fabioh8010 commented on src/pages/workspace/reportFields/EditValuePage.tsx:
```diff
@@ -0,0 +1,98 @@
+    const currentValueName = formDraft?.listValues?.[valueIndex] ?? '';
+
+    const validate = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_REPORT_FIELDS_FORM>) =>
```
```suggestion
        (values: FormOnyxValues<WorkspaceReportFieldsForm>) =>
```

fabioh8010 commented on src/pages/workspace/reportFields/EditValuePage.tsx:
```diff
@@ -0,0 +1,98 @@
+    );
+
+    const editValue = useCallback(
+        (values: FormOnyxValues<typeof ONYXKEYS.FORMS.WORKSPACE_REPORT_FIELDS_FORM>) => {
```
```suggestion
        (values: FormOnyxValues<WorkspaceReportFieldsForm>) => {
```

fabioh8010 commented on src/pages/workspace/reportFields/EditValuePage.tsx:
```diff
@@ -0,0 +1,98 @@
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
```
```suggestion
                style={styles.defaultModalContainer}
```

fabioh8010 commented on src/pages/workspace/reportFields/InitialListValueSelector/InitialListValueSelectorModal.tsx:
```diff
@@ -0,0 +1,91 @@
+            useNativeDriver
+        >
+            <ScreenWrapper
+                style={[styles.pb0]}
```
```suggestion
                style={styles.pb0}
```

fabioh8010 commented on src/pages/workspace/reportFields/InitialListValueSelector/InitialListValueSelectorModal.tsx:
```diff
@@ -0,0 +1,91 @@
+                    onBackButtonPress={onClose}
+                />
+                <View style={[styles.ph5, styles.pb4]}>
+                    <Text style={StyleUtils.combineStyles([styles.sidebarLinkText, styles.optionAlternateText])}>{subtitle}</Text>
```
```suggestion
                    <Text style={StyleUtils.combineStyles([styles.sidebarLinkText, styles.optionAlternateText])}>{subtitle}</Text>
```

Same, do we really need `StyleUtils.combineStyles` here?

fabioh8010 commented on src/pages/workspace/reportFields/ReportFieldTypePicker/index.tsx:
```diff
@@ -0,0 +1,50 @@
+    isSelected: boolean;
+};
+
+type ReportFieldTypePickerProps = {
+    defaultValue: PolicyReportFieldType;
+    onOptionSelected: (reportField: ReportFieldItemType) => void;
+};
```
Please add descriptions to the props

fabioh8010 commented on src/pages/workspace/reportFields/ReportFieldTypePicker/index.tsx:
```diff
@@ -0,0 +1,50 @@
+
+    return (
+        <SelectionList
+            sections={[{data: reportFieldOptions}]}
```
Could we make `reportFieldOptions` return this structure on `useMemo` instead of defining a new array every time we get a render?

fabioh8010 commented on src/pages/workspace/reportFields/TypeSelector/TypeSelectorModal.tsx:
```diff
@@ -0,0 +1,72 @@
+            useNativeDriver
+        >
+            <ScreenWrapper
+                style={[styles.pb0]}
```
```suggestion
                style={styles.pb0}
```

fabioh8010 commented on src/pages/workspace/reportFields/TypeSelector/TypeSelectorModal.tsx:
```diff
@@ -0,0 +1,72 @@
+                    onBackButtonPress={onClose}
+                />
+                <View style={[styles.ph5, styles.pb4]}>
+                    <Text style={StyleUtils.combineStyles([styles.sidebarLinkText, styles.optionAlternateText, styles.pre])}>{subtitle}</Text>
```
Same comment about `StyleUtils.combineStyles`

fabioh8010 commented on src/pages/workspace/reportFields/ValueSettingsPage.tsx:
```diff
@@ -0,0 +1,115 @@
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
```
```suggestion
                style={styles.defaultModalContainer}
```

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceAddValuePage.tsx:
```diff
@@ -0,0 +1,85 @@
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
```
```suggestion
                style={styles.defaultModalContainer}
```

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceCreateReportFieldPage.tsx:
```diff
@@ -0,0 +1,195 @@
+                errors[INPUT_IDS.NAME] = translate('workspace.reportFields.reportFieldNameRequiredError');
+            } else if (Object.values(policy?.fieldList ?? {}).some((reportField) => reportField.name === name)) {
+                errors[INPUT_IDS.NAME] = translate('workspace.reportFields.existingReportFieldNameError');
+            } else if ([...name].length > CONST.WORKSPACE_REPORT_FIELD_POLICY_MAX_LENGTH) {
```
```suggestion
            } else if (name.length > CONST.WORKSPACE_REPORT_FIELD_POLICY_MAX_LENGTH) {
```

rezkiy37 replied:
https://github.com/Expensify/App/pull/44275#discussion_r1658483578

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceCreateReportFieldPage.tsx:
```diff
@@ -0,0 +1,195 @@
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
```
```suggestion
                style={styles.defaultModalContainer}
```

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceListValuesPage.tsx:
```diff
@@ -0,0 +1,271 @@
+
+            return (
+                <ButtonWithDropdownMenu
+                    onPress={() => null}
```
Is this intended?

rezkiy37 replied:
Yes, because I don't need any additional actions. I have handlers for dropdown buttons only. It is a common pattern for this component in the app as well.


https://github.com/Expensify/App/assets/57314631/a3d8fa5a-338d-4333-adec-fab4e661e3bc

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceListValuesPage.tsx:
```diff
@@ -0,0 +1,271 @@
+        >
+            <ScreenWrapper
+                includeSafeAreaPaddingBottom={false}
+                style={[styles.defaultModalContainer]}
```
```suggestion
                style={styles.defaultModalContainer}
```

fabioh8010 commented on src/pages/workspace/reportFields/WorkspaceListValuesPage.tsx:
```diff
@@ -0,0 +1,271 @@
+                {!shouldShowEmptyState && (
+                    <SelectionList
+                        canSelectMultiple
+                        sections={[{data: valueList, isDisabled: false}]}
```
Same, maybe we could make `valueList` return this structure on `useMemo` instead of defining a new array every time

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
clarify comments

rezkiy37 committed:
remove dev. logs

rezkiy37 committed:
remove extra styles array

rezkiy37 committed:
add comment

rezkiy37 committed:
remove extra styles array

rezkiy37 committed:
simplify styles

rezkiy37 committed:
memorize entire sections

rezkiy37 committed:
unify BULK_ACTION_TYPES

rezkiy37 committed:
simplify array styles

rezkiy37 committed:
undo extra changes

fabioh8010 commented on tests/actions/ReportFieldsTest.ts:
```diff
@@ -0,0 +1,94 @@
+        });
+    });
+
+    beforeEach(() => {
+        global.fetch = TestHelper.getGlobalFetchMock();
+        return Onyx.clear().then(waitForBatchedUpdates);
+    });
```
```suggestion
    let mockFetch: MockFetch;
    beforeEach(() => {
        global.fetch = TestHelper.getGlobalFetchMock();
        mockFetch = fetch as MockFetch;
        return Onyx.clear().then(waitForBatchedUpdates);
    });
```

What about this?

rezkiy37 replied:
Done - bd5dacd.

fabioh8010 commented on tests/actions/ReportFieldsTest.ts:
```diff
@@ -0,0 +1,94 @@
+
+    describe('createReportField', () => {
+        it('creates a new text report field of a workspace', async () => {
+            (fetch as MockFetch)?.pause?.();
```
```suggestion
            mockFetch?.pause?.();
```

Then you can use it here and in the other places

rezkiy37 replied:
Done - bd5dacd.

rezkiy37 committed:
create more case scenarios

rezkiy37 committed:
improve test

rezkiy37 committed:
fix en translations

rezkiy37 committed:
add spanish translations

rezkiy37 committed:
handle small screen for header button

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

shawnborton commented:
What's the best way to test this with the adhoc builds above? I'm not seeing a way to enable report fields from More Features.

rezkiy37 commented:
@shawnborton you need a specific beta (`reportFieldsFeature`) to test the flow.

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
improve Spanish translations

shawnborton commented:
Nice, this is looking good! The only minor thing I came across was that this wrapper here in the List values page should have a top padding of 12px, so that there is a bit of space below the page header:
![CleanShot 2024-07-02 at 08 05 54@2x](https://github.com/Expensify/App/assets/2319350/52fd6ce9-6a7d-4308-b6ff-89c4c49058ca)

shawnborton commented:
The button in the List values page slightly jumps/changes places when you use multi-select... can we make sure the button is always in the same spot?
![CleanShot 2024-07-02 at 08 06 56](https://github.com/Expensify/App/assets/2319350/434b7987-c249-4dca-9caa-2b10708e8611)

shawnborton commented:
For the system messages in the #admins room, should we be calling the List value style a "Dropdown" report field? I thought we wanted to change that?
![CleanShot 2024-07-02 at 08 08 38@2x](https://github.com/Expensify/App/assets/2319350/edd89ead-67b8-4304-9708-8cd6bf8e8f97)

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
add top padding

rezkiy37 committed:
fix button layout

rezkiy37 commented:
> For the system messages in the #admins room, should we be calling the List value style a "Dropdown" report field? I thought we wanted to change that? ![CleanShot 2024-07-02 at 08 08 38@2x](https://private-user-images.githubusercontent.com/2319350/344896346-edd89ead-67b8-4304-9708-8cd6bf8e8f97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MTkzODAsIm5iZiI6MTcxOTkxOTA4MCwicGF0aCI6Ii8yMzE5MzUwLzM0NDg5NjM0Ni1lZGQ4OWVhZC02N2I4LTQzMDQtOTcwOC04Y2Q2YmY4ZThmOTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MDJUMTExODAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWFhZDg0YWM3NzhkMWI3MTY4MmEzMTZhNDVlM2MwODQ3ODFhMmE4YmJiZWExNDhhNjVjZTg3YzFhYmYzZTczOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NCckKBdhapM9fB3zoJkojMhI2IW8N_QVDB8Byp5ZGGM)

Regarding this comment, the app does not translate these messages. The text comes from the backend and the app represents it. Also, the app does not generate these messages optimistically. 

cc @mountiny 

<details><summary>Details</summary>

<img width="1409" alt="Screenshot 2024-07-02 at 13 39 36" src="https://github.com/Expensify/App/assets/57314631/1471fa2e-b6d8-4f3a-999f-912fa1a640c7">


</details>

rezkiy37 commented:
@shawnborton, I've fixed (https://github.com/Expensify/App/pull/44275#issuecomment-2202011244, https://github.com/Expensify/App/pull/44275#issuecomment-2202012661). Commits: 805d3b6dae7b2c3f0a01f3b930e2842488f0267a, 6a025909eebbe3c3ecc73391ab949bc2d809eddf.

<details><summary>Details</summary>
<p>


https://github.com/Expensify/App/assets/57314631/d197eba9-3566-4a95-acf5-00a79ddcb460



</p>
</details>

rezkiy37 committed:
add missed types

rezkiy37 committed:
fix report field key

rezkiy37 commented:
> BUG?: Select All option is not working in report fields selection page.



> BUG?: Select All option is not working in report fields selection page.

It is not a bug, because out of the scope of this PR. It is in the scope of https://github.com/Expensify/App/issues/43687.

rezkiy37 committed:
reset selected values on navigate

rezkiy37 committed:
do not clear on success

rezkiy37 commented:
> <img alt="Screenshot 2024-07-02 at 8 21 41 PM" width="1512" src="https://private-user-images.githubusercontent.com/58412969/345084924-7df6b5e7-94a6-4af3-b743-8e6ede14f093.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MzU5MjAsIm5iZiI6MTcxOTkzNTYyMCwicGF0aCI6Ii81ODQxMjk2OS8zNDUwODQ5MjQtN2RmNmI1ZTctOTRhNi00YWYzLWI3NDMtOGU2ZWRlMTRmMDkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE1NTM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5ZDk2OTRkZWQyNDFjMWJkYjI1MDllOTc5YzljOGQxNWNkZjVlZDdkNGY3OWYyOWFlNDY0YjRjZjU1ZjMxNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VvFwN-DCBqnEYu1cxfyLI4rJS8TzvNC7IdD3ArmGfjM">
> BUG: App crashes on selecting optimistic entry
> 
> 1. Go offline.
> 2. Try to add a report field
> 3. Try to select the added report field.
> 4. See the crash



> <img alt="Screenshot 2024-07-02 at 8 21 41 PM" width="1512" src="https://private-user-images.githubusercontent.com/58412969/345084924-7df6b5e7-94a6-4af3-b743-8e6ede14f093.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MzU5MjAsIm5iZiI6MTcxOTkzNTYyMCwicGF0aCI6Ii81ODQxMjk2OS8zNDUwODQ5MjQtN2RmNmI1ZTctOTRhNi00YWYzLWI3NDMtOGU2ZWRlMTRmMDkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE1NTM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5ZDk2OTRkZWQyNDFjMWJkYjI1MDllOTc5YzljOGQxNWNkZjVlZDdkNGY3OWYyOWFlNDY0YjRjZjU1ZjMxNGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VvFwN-DCBqnEYu1cxfyLI4rJS8TzvNC7IdD3ArmGfjM">
> BUG: App crashes on selecting optimistic entry
> 
> 1. Go offline.
> 2. Try to add a report field
> 3. Try to select the added report field.
> 4. See the crash

Fixed - a78985ea67abb0176651db058b5c9b52fac123b9.

rezkiy37 commented:
> <img alt="Screenshot 2024-07-02 at 8 26 16 PM" width="1512" src="https://private-user-images.githubusercontent.com/58412969/345085485-780cdb41-4c09-4231-92d9-dd287c171ab8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MzczNDcsIm5iZiI6MTcxOTkzNzA0NywicGF0aCI6Ii81ODQxMjk2OS8zNDUwODU0ODUtNzgwY2RiNDEtNGMwOS00MjMxLTkyZDktZGQyODdjMTcxYWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE2MTcyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxZDMzZjg4MDc5OGZlMTcwMGFlZmVjOTlmYjg0YjI0OGM3NTkxNmY5ZjQyNzExYTA3MDhhYzE0NWZmNGQwZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8w44eT-oOpGhwB9iP5qLBb094tyj4MYzVcC7o9juVJc">
> BUG: Deleted item doesn't get removed from selection
> 
> 1. In the list view, select the checkbox of the item you want to delete.
> 2. Click on the item to open the item detail page.
> 3. Click on Delete and confirm the delete

Fixed - 05d331908169b110183b7a90cb64efe01f0e586b.

rezkiy37 commented:
> BUG: Noticing some flicker in the report field list while adding the new report fields

Fixed - e005d47.

rezkiy37 committed:
update tests

rezkiy37 committed:
Merge branch 'main' of https://github.com/rezkiy37/Expensify into feature/43684-report-fields-creating-flow-4

rezkiy37 committed:
simplify failureData

rezkiy37 commented on src/libs/actions/Policy/ReportFields.ts:
```diff
@@ -0,0 +1,180 @@
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                onyxMethod: Onyx.METHOD.MERGE,
+                value: {
+                    fieldList: previousFieldList,
```
Simplified - 6f4496b.

rezkiy37 committed:
improved policies onyx connection

rezkiy37 commented on src/libs/actions/Policy/ReportFields.ts:
```diff
@@ -0,0 +1,182 @@
+    key: ONYXKEYS.COLLECTION.POLICY,
+    waitForCollectionCallback: true,
+    callback: (value) => (allPolicies = value),
+});
```
Done - eac25ff.

rezkiy37 committed:
add js doc comments

rezkiy37 commented on src/pages/workspace/reportFields/WorkspaceListValuesPage.tsx:
```diff
@@ -0,0 +1,276 @@
+    value: string;
+    enabled: boolean;
+    orderWeight?: number;
+};
```
Done - f49d243.

rezkiy37 commented on src/pages/workspace/reportFields/ReportFieldTypePicker/index.tsx:
```diff
@@ -0,0 +1,53 @@
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+};
```
Done - f49d243.

mountiny commented on src/pages/workspace/reportFields/WorkspaceAddValuePage.tsx:
```diff
@@ -0,0 +1,85 @@
+    );
+}
+
+WorkspaceAddValuePage.displayName = 'WorkspaceAddValuePage';
```
I hink we should use a better name, is this for creating the initial value for the report field?

rezkiy37 replied:
Yes, but I am considering reusing this screen for the edit flow. So it is adding a new value to the list.

rezkiy37 replied:
What do you think about **WorkspaceAddListValuePage**?

mountiny replied:
`ReportFieldAddListValuePage` I think will be the best as we are not reusing it for other features than the report fields

rezkiy37 replied:
I've renamed a few components:

1. **WorkspaceAddValuePage** -> **ReportFieldAddListValuePage** (15f33ea84d0354aa96cc99e191c09fe0971074fd)
2. **WorkspaceListValuesPage** -> **ReportFieldListValuesPage** (64fca652a1ff11551c541e589286416388f564a4)
3. **EditValuePage** -> **ReportFieldEditValuePage** (256ca29af11a999f948a8dfa12d55ed6b752b405)

cc @shubham1206agra

mountiny commented on tests/actions/ReportFieldsTest.ts:
```diff
@@ -0,0 +1,237 @@
+            expect(policy?.pendingFields?.[reportFieldKey]).toBeFalsy();
+        });
+
+        it('creates a new list report field of a workspace', async () => {
```
@rezkiy37 can you please add Edit and delete flows to the unit tests in a follow up PR? Thanks

rezkiy37 replied:
Yes, I will do it.

rezkiy37 committed:
rename page add value page

rezkiy37 committed:
rename list values page

rezkiy37 committed:
rename edit value page

mountiny commented on src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx:
```diff
@@ -361,6 +361,11 @@ const SettingsModalStackNavigator = createModalStackNavigator<SettingsNavigatorP
     [SCREENS.SETTINGS.SUBSCRIPTION.CHANGE_BILLING_CURRENCY]: () => require<ReactComponentModule>('../../../../pages/settings/Subscription/PaymentCard/ChangeBillingCurrency').default,
     [SCREENS.SETTINGS.SUBSCRIPTION.ADD_PAYMENT_CARD]: () => require<ReactComponentModule>('../../../../pages/settings/Subscription/PaymentCard').default,
     [SCREENS.SETTINGS.ADD_PAYMENT_CARD_CHANGE_CURRENCY]: () => require<ReactComponentModule>('../../../../pages/settings/PaymentCard/ChangeCurrency').default,
+    [SCREENS.WORKSPACE.REPORT_FIELDS_CREATE]: () => require<ReactComponentModule>('../../../../pages/workspace/reportFields/WorkspaceCreateReportFieldPage').default,
```
I think we could remove the Workspace from the name here as well, but that can be done in next PR
```suggestion
    [SCREENS.WORKSPACE.REPORT_FIELDS_CREATE]: () => require<ReactComponentModule>('../../../../pages/workspace/reportFields/CreateReportFieldPage').default,
```

rezkiy37 replied:
Yes, will do it 🙂

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9783702773) to staging by https://github.com/mountiny in version: 9.0.4-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/9800117961) to staging by https://github.com/tgolen in version: 9.0.4-5 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9872741083) to production by https://github.com/Julesssss in version: 9.0.5-13 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9945901401) to production by https://github.com/thienlnam in version: 9.0.6-8 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

rezkiy37 commented on src/libs/actions/Policy/ReportFields.ts:
```diff
@@ -0,0 +1,209 @@
+        fieldID,
+        orderWeight: Object.keys(previousFieldList).length + 1,
+        deletable: false,
+        value: type === CONST.REPORT_FIELD_TYPES.LIST ? CONST.REPORT_FIELD_TYPES.LIST : null,
```
Alright, I got it.

mountiny replied:
I checked and seems like its `null` even in oldDot when its dropdown type. So I think you can explore removing it
